### DISCUSSION TITLE
خطای resolve service

### DISCUSSION THREAD

**Discussion by Mr-Mohammad** [Sina-Soltani/Parbad#360] [Q&A]
سلام من از .net core استفاده میکنم و در فایل Program.cs کانفیگ پرباد به این صورت نوشتم 
```
builder.Services.AddParbad()
.ConfigureGateways(gateways =>
{
    gateways
         .AddMellat()
         .WithAccounts(source =>
         {
             source.AddInMemory(account =>
             {
                 account.TerminalId = 123;
                 account.UserName = "MyId";
                 account.UserPassword = "MyPassword";
             });
         });
}).Build();
```
هنگام Run Time  این خطا رخ میده
System.AggregateException: 'Some services are not able to be constructed (Error while validating the service descriptor 'ServiceType: Parbad.IOnlinePayment Lifetime: Transient ImplementationType: Parbad.Internal.DefaultOnlinePayment': Unable to resolve service for type 'Parbad.Storage.Abstractions.IStorageManager' while attempting to activate 'Parbad.Internal.DefaultOnlinePayment'.) (Error while validating the service descriptor 'ServiceType: Parbad.TrackingNumberProviders.AutoIncrementTrackingNumber Lifetime: Transient ImplementationType: Parbad.TrackingNumberProviders.AutoIncrementTrackingNumber': Unable to resolve service for type 'Parbad.Storage.Abstractions.IStorage' while attempting to activate 'Parbad.TrackingNumberProviders.AutoIncrementTrackingNumber'.) (Error while validating the service descriptor 'ServiceType: Parbad.TrackingNumberProviders.AutoRandomTrackingNumber Lifetime: Transient ImplementationType: Parbad.TrackingNumberProviders.AutoRandomTrackingNumber': Unable to resolve service for type 'Parbad.Storage.Abstractions.IStorage' while attempting to activate 'Parbad.TrackingNumberProviders.AutoRandomTrackingNumber'.)'

**Comment by Sina-Soltani** [comment]
درود
نمونه کدها، مثال‌ها و آموزش‌ها کاملا واضح هست. لطفا نگاهی به اونها بندازید.
متد ConfigureGateway رو استفاده نکردید
